pandas/core/algorithms.py

```diff
@@ -2070,20 +2070,38 @@ def sort_mixed(values):
         strs = np.sort(values[str_pos])
         return np.concatenate([nums, np.asarray(strs, dtype=object)])
 
+    def sort_tuples(values: np.ndarray[tuple]):
```

Comment:
can you move sort_tuples and sort_mixed to module level, rename as private (e.g. _sort_tuples) and put right after safe_sort.

Reply:
done